api/v1beta1/clusterclass_types.go

```diff
@@ -583,6 +583,42 @@ type JSONSchemaProps struct {
 	// It can be used to add additional data for higher level tools.
 	// +optional
 	XMetadata *VariableSchemaMetadata `json:"x-metadata,omitempty"`
+
+	// AllOf specifies that the variable must validate against all of the subschemas in the array.
```

Comment:
I don't see any implementation for these fields in clusterclass_variable_validation.go validateSchema.

I would expect us to go through the recursion like for other recursive schemas like e.g. "items".

(upstream reference: https://github.com/kubernetes/kubernetes/blob/a5791b344c04ded4f443c7e134242a29bd0e2bac/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go#L1056-L1077)

(+ corresponding unit test coverage)

Reply:
Added validation to validateSchema. Still need to add tests.

Reply:
Added some tests, not happy with coverage but also struggling to think of other test cases to add without re-covering upstream validation too much.